test/inductor/test_torchinductor.py

```diff
@@ -5128,6 +5122,199 @@ def fn(x):
+    "test_sizehint_issue1_dynamic_shapes": ("cpu", "cuda"),
+    "test_slice1_dynamic_shapes": ("cuda",),
+    "test_slice2_dynamic_shapes": ("cuda",),
+    "test_slice_mutation1_dynamic_shapes": ("cuda",),
+    "test_slice_scatter_dynamic_shapes": ("cuda",),
+    "test_softmax_dynamic_shapes": ("cuda",),
+    "test_softmax_one_kernel_dynamic_shapes": ("cuda",),
+    "test_split_with_sizes_dynamic_shapes": ("cuda",),
+    "test_squeeze2_dynamic_shapes": ("cuda",),
+    "test_std_dynamic_shapes": ("cuda",),
+    "test_strided_inputs_dynamic_shapes": ("cpu", "cuda"),
+    "test_sum1_dynamic_shapes": ("cuda",),
+    "test_sum2_dynamic_shapes": ("cuda",),
+    "test_sum3_dynamic_shapes": ("cuda",),
+    "test_sum4_dynamic_shapes": ("cuda",),
+    "test_sum5_dynamic_shapes": ("cuda",),
+    "test_sum_dtype_dynamic_shapes": ("cuda",),
+    "test_sum_keepdims_dynamic_shapes": ("cuda",),
+    "test_tanh_dynamic_shapes": ("cuda",),
+    "test_tmp_not_defined_issue1_dynamic_shapes": ("cuda",),
+    "test_tmp_not_defined_issue2_dynamic_shapes": ("cpu", "cuda"),
+    "test_to_memory_format_dynamic_shapes": ("cuda",),
+    "test_transpose_add_dynamic_shapes": ("cuda",),
+    "test_transpose_dynamic_shapes": ("cuda",),
+    "test_transposed_propagates_dynamic_shapes": ("cuda",),
+    "test_triu_dynamic_shapes": ("cuda",),
+    "test_unroll_small_reduction_dynamic_shapes": ("cpu", "cuda"),
+    "test_unspec_inputs_dynamic_shapes": ("cpu", "cuda"),
+    "test_unsqueeze_dynamic_shapes": ("cuda",),
+    "test_unsqueeze_inplace_dynamic_shapes": ("cuda",),
+    "test_upsample_bilinear2d_a_dynamic_shapes": ("cpu", "cuda"),
+    "test_upsample_bilinear2d_b_dynamic_shapes": ("cpu", "cuda"),
+    "test_upsample_nearest1d_dynamic_shapes": ("cpu", "cuda"),
+    "test_upsample_nearest2d_backward_dynamic_shapes": ("cpu", "cuda"),
+    "test_upsample_nearest2d_dynamic_shapes": ("cpu", "cuda"),
+    "test_upsample_nearest3d_dynamic_shapes": ("cpu", "cuda"),
+    "test_var_mean_dynamic_shapes": ("cuda",),
+    "test_vertical_fusion1_dynamic_shapes": ("cuda",),
+    "test_views1_dynamic_shapes": ("cuda",),
+    "test_views3_dynamic_shapes": ("cpu",),
+}
+
+
+def copy_tests(my_cls, other_cls, suffix):  # noqa: B902
+    for name, value in my_cls.__dict__.items():
+        if name.startswith("test_"):
+            # You cannot copy functions in Python, so we use lambdas here to
+            # create objects with different ids. Otherwise, unittest.skip
+            # would modify all methods sharing the same object id. Also, by
+            # using a default argument in a lambda, we create a copy instead of
+            # a reference. Otherwise, we would lose access to the value.
+            skips = test_skips.get(name)
+            if skips and suffix in skips:
+                setattr(
+                    other_cls,
+                    f"{name}_{suffix}",
+                    unittest.skip("Skipped!")(lambda self, value=value: value(self)),
```

Comment:
should it be skip or xfail?

Reply:
I had xfails initially, then changed to skips because of inconsistent behavior between machines/envs.

Some tests pass on my dev box, but fail on CI. Some only fail ASAN tests on CI, but pass otherwise. I think it'll be very hard to maintain xfailures as of now, so let's just slowly work on enabling each op instead.

By landing this sooner, we'll at least have some baseline. Right now, we have no dyn-shape tests on CI at all.